src/libmongoc/tests/unified/operation.c

```diff
@@ -1452,6 +1535,29 @@ operation_iterate_until_document_or_error (test_t *test,
    return ret;
 }
 
+static bool
+operation_close (test_t *test,
+                 operation_t *op,
+                 result_t *result,
+                 bson_error_t *error)
+{
+   bool ret = false;
+   mongoc_cursor_t *cursor = NULL;
+
+   cursor = entity_map_get_cursor (test->entity_map, op->object, error);
+   if (!cursor) {
+      goto done;
+   }
+
+   entity_map_delete (test->entity_map, op->object, error);
```

Comment:
This borrowed logic from [PHPLIB-669](https://jira.mongodb.org/browse/PHPLIB-669) where deleting the Cursor and triggering garbage collection was the only way for us to invoke `mongoc_cursor_destroy`. libmongoc doesn't have the same technical limitation. Is this preferable to calling `_mongoc_client_kill_cursor` (since that's still a private API)?

I suppose there's no value in having the cursor entity stick around otherwise. Perhaps it'd be good to clarify this in the unified spec: [DRIVERS-1841](https://jira.mongodb.org/browse/DRIVERS-1841)

Reply:
I was considering this, but decided that `entity_map_delete` is the preferred way in C as well:
* We don't know what kind of cursor implementation backs the `mongoc_cursor_t` - they each have their own destruction logic, which can only be triggered through `mongoc_cursor_destroy`
* Calling `mongoc_cursor_destroy` destroys everything related to the cursor, meaning that we shouldn't be holding a reference to it anymore, hence the `entity_map_delete`.

If @kevinAlbs thinks it's safe to always call `killCursors` and leave the (now closed) cursor in the entity map, I suppose using `_mongoc_client_kill_cursor` would be fine.

Reply:
Good point about different cursor implementations. I'm fine with keeping this as-is.